crates/networking/p2p/rlpx/message.rs

```diff
@@ -201,8 +209,10 @@ impl Message {
             Message::ByteCodes(msg) => msg.encode(buf),
             Message::GetTrieNodes(msg) => msg.encode(buf),
             Message::TrieNodes(msg) => msg.encode(buf),
-            Message::NewBlock(msg) => msg.encode(buf),
-            Message::BatchSealed(msg) => msg.encode(buf),
+            Message::L2Message(l2_msg) => match l2_msg {
+                L2Message::BatchSealed(msg) => msg.encode(buf),
+                L2Message::NewBlock(msg) => msg.encode(buf),
+            },
```

Comment:
Can this be expressed like this?

```suggestion
            Message::L2Message(L2Message::NewBlock(msg)) => msg.encode(buf),
            Message::L2Message(L2Message::BatchSealed(msg)) => msg.encode(buf),
```

Reply:
Yes, it could be expressed like that, but I find the inner match approach easier to read..